controllers/flowlogspipeline/flp_monolith_reconciler.go

```diff
@@ -94,18 +95,28 @@ func (r *flpMonolithReconciler) reconcile(ctx context.Context, desired *flowslat
 	}
 
 	builder := newMonolithBuilder(r.nobjMngr.Namespace, r.image, &desired.Spec, r.useOpenShiftSCC, r.CertWatcher)
-	newCM, configDigest, err := builder.configMap()
+	newCM, configDigest, dbConfigMap, err := builder.configMap()
 	if err != nil {
 		return err
 	}
 	if !r.nobjMngr.Exists(r.owned.configMap) {
 		if err := r.CreateOwned(ctx, newCM); err != nil {
 			return err
 		}
+		if r.reconcilersCommonInfo.availableAPIs.HasConsoleConfig() {
+			if err := r.CreateOwned(ctx, dbConfigMap); err != nil {
+				return err
+			}
+		}
 	} else if !equality.Semantic.DeepDerivative(newCM.Data, r.owned.configMap.Data) {
 		if err := r.UpdateOwned(ctx, r.owned.configMap, newCM); err != nil {
 			return err
 		}
+		if r.reconcilersCommonInfo.availableAPIs.HasConsoleConfig() {
+			if err := r.UpdateOwned(ctx, r.owned.dbConfigMap, dbConfigMap); err != nil {
```

Comment:
not sure about that ... basically it is saying : "if the main configMap is changed, then update the DB config map" ?
I would rather copy-paste the whole if/else block and adapt it for dbConfiogMap

Reply:
(same applies to `flp_transfo_reconciler` file)

Reply:
If the main configMap is changed, it is possible that the labels in the dashboard are changed because of a new prefix to the prometheus variables. Otherwise, there is no reason to track the changes to the new dashboards configMap, and as noted above, the dashboard configMap does not technically belong to the netobserv` namespace.

Reply:
yeah, but tying the `dbConfigMap` lifecycle to the presence of `configMap` seems risky to me

For instance, imagine that thr process runs through the top `if !r.nobjMngr.Exists(r.owned.configMap)` at first creation, and `configMap` is created, so far so good. Then imagine there's an issue (e.g. a temporary connectivity issue) and the subsequent call to `r.CreateOwned(ctx, dbConfigMap)` fails: in that case,  `configMap` exists and `dbConfigMap`. In such scenario, the operator will never be able to recover from that state, as the reconcile doesn't create `dbConfigMap` when `configMap` exists.

Reply:
The worst that happens is that there is no dashbaord, but everything else runs properly.

To manage it independently from the original configMap, we have to track it. We cannot put it in the `netobserv` nobjMngr structure because it is not in the `netobserv` namespace and it would cause other confusion. So we would need another place to save the status of this particular dbConfigMap. Where would you suggest?

Reply:
Yes, I agree. Something like this should work I guess:

```go
func (r *reconcilersCommonInfo) reconcileDbConfig(ctx context.Context, dbConfigMap *corev1.ConfigMap) error {
	curr := &corev1.ConfigMap{}
	if err := r.Get(ctx, types.NamespacedName{
		Name:      dbConfigMap.Name,
		Namespace: dbConfigMap.Namespace,
	}, curr); err != nil {
		if errors.IsNotFound(err) {
			return r.CreateOwned(ctx, dbConfigMap)
		}
		return err
	}
	if !equality.Semantic.DeepDerivative(dbConfigMap.Data, curr.Data) {
		return r.UpdateOwned(ctx, curr, dbConfigMap)
	}
	return nil
}
```

(Same function can be shared between monolith & transfo reconcilers)
It makes also the `dbConfigMap:    &corev1.ConfigMap{}` in `owned` struct useless

Reply:
done